### PR TITLE
Enable nullability in ExpressionContext and ExpressionTable

### PR DESCRIPTION
**Pull request by halgab**
Contributes to #1107

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8776)

### CODE REVIEW

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/ExpressionTable.cs [hunk 1]
```diff
@@ -13,7 +11,7 @@ namespace System.ComponentModel.Design.Serialization
     /// </summary>
     internal sealed class ExpressionTable
     {
-        private Dictionary<object, ExpressionInfo> _expressions;
+        private Dictionary<object, ExpressionInfo>? _expressions;
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/ExpressionTable.cs:14]: I do not see need for a backing field as well as property on it. Can you simplify the code to use only one of them and after that, we don't need it nullable.

File: src/System.Windows.Forms.Design/src/PublicAPI.Shipped.txt [hunk 2]
```diff
@@ -298,12 +298,12 @@
...
 ~System.ComponentModel.Design.Serialization.CodeDomSerializerException.LinePragma.get -> System.CodeDom.CodeLinePragma
 ~System.ComponentModel.Design.Serialization.CollectionCodeDomSerializer.MethodSupportsSerialization(System.Reflection.MethodInfo method) -> bool
-~System.ComponentModel.Design.Serialization.ExpressionContext.Expression.get -> System.CodeDom.CodeExpression
-~System.ComponentModel.Design.Serialization.ExpressionContext.ExpressionContext(System.CodeDom.CodeExpression expression, System.Type expressionType, object owner) -> void
-~System.ComponentModel.Design.Serialization.ExpressionContext.ExpressionContext(System.CodeDom.CodeExpression expression, System.Type expressionType, object owner, object presetValue) -> void
-~System.ComponentModel.Design.Serialization.ExpressionContext.ExpressionType.get -> System.Type
-~System.ComponentModel.Design.Serialization.ExpressionContext.Owner.get -> object
-~System.ComponentModel.Design.Serialization.ExpressionContext.PresetValue.get -> object
+System.ComponentModel.Design.Serialization.ExpressionContext.Expression.get -> System.CodeDom.CodeExpression
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/src/PublicAPI.Shipped.txt:301]: You are not supposed to update this file manually. Just build your changes locally and build will update `Unshipped.txt`.

**Review by dreddy-work — CHANGES_REQUESTED** [review]
please check build failure.

**Comment by halgab** [comment]
Well, I thought my second commit would fix them, and I'm not really sure why it hasn't...

File: src/System.Windows.Forms.Design/src/PublicAPI.Shipped.txt (hunk 2 above)
Comment (dreddy-work) [reply src/System.Windows.Forms.Design/src/PublicAPI.Shipped.txt:301]: May be not build, but you should see Analyzer suggestion to add it to the API's.

We track these and move to Shipped when ready for current preview release.

Comment (halgab) [reply src/System.Windows.Forms.Design/src/PublicAPI.Shipped.txt]: Okay, I saw the analyser this time. Thanks for your help!

**Comment by dreddy-work** [comment]
@halgab, when you get chance, can you rebase it on main?

**Comment by dreddy-work** [comment]
Seems conflict is still not resolved. @halgab.

**Comment by halgab** [comment]
should be good now, sorry

**Review by dreddy-work — APPROVED** [review]

**Pull request merged**